src/search/landmarks/landmark_factory.h

```diff
@@ -38,70 +33,53 @@ class LandmarkFactory {
     virtual bool supports_conditional_effects() const = 0;
 
 protected:
+    explicit LandmarkFactory(const options::Options &opts);
+
     std::shared_ptr<LandmarkGraph> lm_graph;
-    AbstractTask *lm_graph_task;
+    const bool reasonable_orders;
+    const bool only_causal_landmarks;
+    const bool disjunctive_landmarks;
+    const bool conjunctive_landmarks;
+    const bool no_orders;
 
+    // TODO: Directly use no_orders in LandmarkFactoryHM and remove this
```

Comment:
I'm not sure I agree with the TODO. If we can make data private, we are usually happy. But this is probably gone anyway with the other changes we will do.